docs/build/isc/v1.1/docs/tutorials/defi-yield-farming.md

```diff
@@ -0,0 +1,422 @@
+
+        /// Ensure that rewards can only be issued once every 60 minutes
+        require(
+            block.timestamp - lastReward > 60 minutes,
+            "Need to wait 60 minutes"
+        );
+
+        /// Fetch the current IOTA/USD price from the Pyth Oracle.
+        uint256 iotaToUSDPrice = getIotaToUsdPriceFeed(priceUpdate);
+
+
+        /// Loop through all stakers and distribute rewards based on their staked amount
+        for (uint256 i = 0; i < stakers.length; i++) {
+            address recipient = stakers[i]; // Get the address of the staker
+            uint256 balance = stakingBalance[recipient]; // Get the staker's balance
+
+            /// If the staker has a non-zero balance, issue rewards
+            if (balance > 0) {
+                /// Calculate the reward based on the staked value in USD
+                uint256 stakedBalanceinUSD = (balance * iotaToUSDPrice) /
+                    10 ** 18;
+
+                /// Determine reward amount, e.g., 5% of the staked value in USD
+                uint256 reward = (stakedBalanceinUSD * 5) / 100;
+
+                /// Transfer the reward to the staker
+                dappToken.transfer(
+                    recipient,
+                    reward * 10 ** dappToken.decimals()
+                );
+            }
+        }
+
+        /// Update the timestamp of the last reward issuance
+        lastReward = block.timestamp;
+    }
+}
+```
+
+## Step 3: Deployment Script (HardHat)
+
+Create a deployment script to deploy the contracts to the IOTA EVM testnet.
+
+In the `scripts` folder, create a file named `deploy.js`:
+
+```javascript
+async function main() {
+    /// Fetch contract factories
+    const DappToken = await ethers.getContractFactory("DappToken");
+    const IOTAToken = await ethers.getContractFactory("IOTAToken");
+    const TokenFarm = await ethers.getContractFactory("TokenFarm");
+
+    /// Deploy contracts
+    const dappToken = await DappToken.deploy();
+    await dappToken.deployed();
+    console.log("DappToken deployed to:", dappToken.address);
+
+    const iotaToken = await IOTAToken.deploy();
+    await iotaToken.deployed();
+    console.log("IOTAToken deployed to:", iotaToken.address);
+
+    const tokenFarm = await TokenFarm.deploy(dappToken.address, iotaToken.address, "PYTH_ORACLE_CONTRACT_ADDRESS");
+    await tokenFarm.deployed();
+    console.log("TokenFarm deployed to:", tokenFarm.address);
+}
+
+main()
+    .then(() => process.exit(0))
+    .catch((error) => {
+        console.error(error);
+        process.exit(1);
+    });
+
+```
+
+## Step 4: Deploying to the IOTA EVM Test
+
+### 1. Configure Hardhat for IOTA EVM Testnet
+
+Create a `.env` file with the following content (replace with your own private key and RPC URL):
+
+```bash
+PRIVATE_KEY=your-private-key
+IOTA_RPC_URL=https://json-rpc.evm.testnet.iotaledger.net
+```
+
+### 2. Install the dotenv package to manage environment variables:
```

Comment:
```suggestion
### 2. Install the `dotenv` Package to Manage Environment Variables
```

Why not do this when installing the project deps?

Reply:
Wanted this process to be implemented when about to deploy the contract.